src/triton_gazebo/synchronized_image_saver.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+import rclpy
+from rclpy.node import Node
+import os
+from std_msgs.msg import String
+from sensor_msgs.msg import Image
+from triton_interfaces.msg import DetectionBox
+import cv_bridge
+from cv2 import cv2
+import numpy as np
+import random
+
+class SynchronizedImageSaver(Node):
+
+    def __init__(self):
+        super().__init__('synchronized_image_saver')
+        self.subscribe_image = self.create_subscription(Image, "/triton/gazebo_drivers/front_camera/underwater/image_raw", self.save_image, 10)
+        self.subscriber_bbox = self.create_subscription(DetectionBox, "/triton/gazebo_drivers/front_camera/bounding_box", self.save_bbox, 10)
+
+    def save_image(self, msg: Image):
+        self.get_logger().info("Getting image...")
+        br = cv_bridge.CvBridge()
+        im = br.imgmsg_to_cv2(msg, "passthrough")
+        self.current_image = im
+
+    def save_bbox(self, msg: DetectionBox):
+        if (msg.width <= 0 or msg.height <= 0):
+            self.get_logger().info("Bounding box has zero size.")
+            return
+
+        try:
+            rows,cols,channels = self.current_image.shape
+           
+            centre_x = (msg.x + msg.width/2)/cols
+            centre_y = (msg.y + msg.height/2)/rows
+            width = msg.width/cols
+            height = msg.height/rows
+
+            name = "image"+str(random.randint(0,2**16-1))
+            self.get_logger().info("Saving..."+name)
+            txt_string = f"{msg.class_id} {centre_x} {centre_y} {width} {height}"
+            f = open(os.getcwd()+"/src/triton_gazebo/data/" + name + ".txt", "w")
```

Comment:
I believe this will fail if you run the node anywhere else than from the triton folder as the cwd of a launch will be where you ran the launch command. Maybe try using the get_package_share_directory() function https://github.com/ament/ament_index/blob/master/ament_index_python/ament_index_python/packages.py note that the images will be saved in the install folder but it will work on every machine from any directory